addons/pos_online_payment_self_order/static/src/pages/cart_page/cart_page.js

```diff
@@ -39,17 +39,30 @@ patch(CartPage.prototype, {
                 this.checkAndOpenPaymentPage(order);
             }
         } else if (mode === "each") {
-            this.sendInProgress = true;
-            const order = await this.selfOrder.sendDraftOrderToServer();
-            this.sendInProgress = false;
             this.checkAndOpenPaymentPage(order);
         } else {
             return super.pay(...arguments);
         }
     },
-    checkAndOpenPaymentPage(order) {
+    async checkAndOpenPaymentPage(order) {
+        const isOnlinePayment = this.selfOrder.pos_payment_methods.find((p) => p.is_online_payment);
+
         if (order) {
             if (order.state === "draft") {
+                if (!isOnlinePayment) {
+                    this.selfOrder.notification.add(
+                        _t("The current order cannot be paid (no online payment method)."),
+                        { type: "danger" }
+                    );
+                    return;
+                }
+
+                if (!order.isSavedOnServer) {
+                    this.sendInProgress = true;
```

Comment:
I think `sendInProgress` needs to be used in `sendDraftOrderToServer`. However, I don't see it being used. Maybe `sendInProgress` is not needed anymore? If so, restore it's use.